src/js/containers/sharedContainers/LocationListContainer.jsx

```diff
@@ -0,0 +1,102 @@
+/**
+* LocationListContainer.jsx
+* Created by Emily Gullo 12/1/2016
+**/
+
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+
+import PoPTypeahead from 'components/search/filters/location/PlaceOfPerformanceTypeahead';
+
+import * as SearchHelper from 'helpers/searchHelper';
+import * as searchFilterActions from '../../redux/actions/search/searchFilterActions';
+
+const propTypes = {
+    setAutocompleteLocations: React.PropTypes.func,
+    selectLocation: React.PropTypes.func,
+    selectedLocations: React.PropTypes.object,
+    locationOption: React.PropTypes.string
+};
+
+class LocationListContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.handleTextInput = this.handleTextInput.bind(this);
+    }
+
+    dataFormatter(item) {
+        let itemLabel = `<strong>${item.place}</strong><br>${_.upperCase(item.place_type)}`;
+        if (item.parent !== null) {
+            itemLabel += ` in ${item.parent}`;
+        }
+
+        return {
+            label: itemLabel,
+            value: item.place
+        };
+    }
+
+    handleTextInput(locationInput) {
```

Comment:
Can we add 2 features to this:

1. There should be a timeout before a new API call goes out. Basically, if the user types really fast, we don't need to send out an API request on each keystroke since these are "expensive"; we can wait a few ms for the user to calm down first. Some timeout logic does exist in the child PlaceOfPerformanceTypeahead container, but that only deals with showing and hiding length error messages.

2. Even with a timeout, we need to ensure that we cancel any in-flight requests before starting a new one. API calls aren't guaranteed to finish in the order they were made, so an earlier request could return _after_ a later request, resulting in nonsensical data and wiping out the better data. For example, the user types `san `, an API call goes out (API call 1). The user keeps typing `san f`, second API call goes out (API call 2). If API call 2 comes back before API call 1, API call 1's callback/promise chain will cause the autocomplete array in Redux to be nuked and populated with API call 1's results (which may include results that no longer make sense, like `San Diego` and `San Antonio`).